apiserver/pkg/util/cluster.go

```diff
@@ -800,14 +822,20 @@ func (c *RayCluster) SetAnnotationsToAllTemplates(key string, value string) {
 
 // Build compute template
 func NewComputeTemplate(runtime *api.ComputeTemplate) (*corev1.ConfigMap, error) {
+	extendedResourcesJSON, err := json.Marshal(runtime.ExtendedResources)
+	if err != nil {
+		return nil, fmt.Errorf("failed to marshal extended resources: %v", err)
```

Comment:
When we fail to marshal runtime.Tolerations on line 842, we log error instead and leave tolerations unset. Should we consider something similar for extended resources?

Reply:
Returning is probably better here, we should consider updating line 841 below to also return error in a follow-up PR

Reply:
In this change, I've modified the `NewComputeTemplate` function to return an error when marshaling `runtime.Tolerations` fails: https://github.com/ray-project/kuberay/pull/2444